### PR TITLE
fix memory leak

### PR DESCRIPTION
**Pull request by jakekeeys**

### CODE REVIEW

**Comment by mjgarton** [comment]
Nice.

Any reason this fix isn't upstream?

**Comment by mjgarton** [comment]
Also, did you measure this as the problem, or find it by code inspection?

The reason I ask is that I don't see connections leaking at runtime when checking via netstat.

**Comment by jakekeeys** [comment]
still investigating... will merge upstream when the fix is confirmed and https://grafana.dev.uw.systems/d/N3bp9akik/proximo?refresh=1m&panelId=4&fullscreen&edit&orgId=1&var-namespace=customer-platform&var-interval=10m&tab=axes

**Comment by mjgarton** [comment]
Yes I've been watching that too, but not seeing connections going up.  The memory seems to increase even though no connections are closing or new ones being opened, as far as I can see.  Wondering about exposing memory profiling pprof bits. What do you think?

File: proximo-server/nats.go [hunk 1]
```diff
@@ -41,8 +40,8 @@ func (h *natsHandler) HandleConsume(ctx context.Context, consumer, topic string,
 }
 
 func (h *natsHandler) HandleProduce(ctx context.Context, topic string, forClient chan<- *Confirmation, messages <-chan *Message) error {
-
 	conn, err := nats.Connect(h.url)
+	defer conn.Close()
```
Comment (mjgarton) [inline proximo-server/nats.go]: I suspect this will panic if conn is null, which it will be if err != nil

File: proximo-server/nats.go [hunk 2]
```diff
@@ -67,6 +66,7 @@ func (h *natsHandler) HandleProduce(ctx context.Context, topic string, forClient
 
 func (h *natsHandler) Status() (bool, error) {
 	conn, err := nats.Connect(h.url)
+	defer conn.Close()
```
Comment (mjgarton) [inline proximo-server/nats.go]: same here

File: proximo-server/nats-streaming.go [hunk 3]
```diff
@@ -19,12 +19,11 @@ type natsStreamingHandler struct {
 }
 
 func (h *natsStreamingHandler) HandleConsume(ctx context.Context, consumer, topic string, forClient chan<- *Message, confirmRequest <-chan *Confirmation) error {
-
 	conn, err := stan.Connect(h.clusterID, consumer+generateID(), stan.NatsURL(h.url))
+	defer conn.Close()
```
Comment (mjgarton) [inline proximo-server/nats-streaming.go]: I suspect this will panic if conn is null, which it will be if err != nil

**Comment by jakekeeys** [comment]
that's the next port of call if this has no effect

**Pull request closed without merging**